homeassistant/components/sonos/media_player.py

```diff
@@ -255,6 +257,23 @@ def __init__(self, speaker: SonosSpeaker) -> None:
         self._attr_unique_id = self.soco.uid
         self._attr_name = self.speaker.zone_name
 
+    async def async_added_to_hass(self) -> None:
+        """Handle common setup when added to hass."""
+        await super().async_added_to_hass()
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{SONOS_MEDIA_UPDATED}",
+                self.async_write_media_state,
+            )
+        )
+
+    @callback
+    def async_write_media_state(self, uid: str) -> None:
+        """Write media state if the provided UID is coordinator of this speaker."""
+        if self.coordinator.uid == uid:
```

Comment:
Can't we just subscribe to this coordinator instead of listening to a dispatcher signal?